Analysis/Tasks/PWGHF/taskD0D0barCorrelation.cxx

```diff
@@ -0,0 +1,823 @@
+     {"hDeltaPhiVsPtStdPhi", "D0,D0bar candidates;#it{p}_{T}^{D0};#it{p}_{T}^{D0bar};#it{#varphi}^{D0bar}-#it{#varphi}^{D0};entries", {HistType::kTH3F, {{100, 0., 10.}, {100, 0., 10.}, {128, -o2::constants::math::PI / 2., 3. * o2::constants::math::PI / 2.}}}},
+     {"hDeltaPhiVsPtByVtxPhi", "D0,D0bar candidates;#it{p}_{T}^{D0};#it{p}_{T}^{D0bar};#it{#varphi}^{D0bar}-#it{#varphi}^{D0};entries", {HistType::kTH3F, {{100, 0., 10.}, {100, 0., 10.}, {128, -o2::constants::math::PI / 2., 3. * o2::constants::math::PI / 2.}}}}}};
+
+  Configurable<int> d_selectionFlagD0{"d_selectionFlagD0", 1, "Selection Flag for D0"};
+  Configurable<int> d_selectionFlagD0bar{"d_selectionFlagD0bar", 1, "Selection Flag for D0bar"};
+  Configurable<double> cutEtaCandMax{"cutEtaCandMax", -1., "max. cand. pseudorapidity"};
+  Configurable<double> cutPtCandMin{"cutPtCandMin", -1., "min. cand. pT"};
+
+  Filter filterSelectCandidates = (aod::hf_selcandidate_d0::isSelD0 >= d_selectionFlagD0 || aod::hf_selcandidate_d0::isSelD0bar >= d_selectionFlagD0bar);
+
+  void process(aod::Collision const& collision, soa::Filtered<aod::Big2ProngMC> const& candidates, aod::McParticles const& particlesMC, aod::BigTracksMC const& tracksMC)
+  {
+    for (auto& candidate1 : candidates) {
+      //check decay channel flag for candidate1
+      if (!(candidate1.hfflag() & 1 << D0ToPiK)) {
+        continue;
+      }
+      if (cutEtaCandMax >= 0. && std::abs(candidate1.eta()) > cutEtaCandMax)
+        continue;
+      if (cutPtCandMin >= 0. && std::abs(candidate1.pt()) < cutPtCandMin)
+        continue;
+      registry.fill(HIST("hmass"), InvMassD0(candidate1));
+      registry.fill(HIST("hEta"), candidate1.eta());
+
+      //D-Dbar correlation dedicated section
+      //if it's a candidate D0, search for D0bar and evaluate correlations
+      if (candidate1.isSelD0() >= d_selectionFlagD0) {
+        double xPrimaryVertex = candidate1.index0_as<aod::BigTracksMC>().collision().posX(), yPrimaryVertex = candidate1.index0_as<aod::BigTracksMC>().collision().posY();
+        double pt1 = candidate1.pt(), phi1Std = candidate1.phi();
+        double phi1ByVtx = evaluatePhiByVertex(xPrimaryVertex, candidate1.xSecondaryVertex(), yPrimaryVertex, candidate1.ySecondaryVertex());
+        registry.fill(HIST("hPhiStdPhi"), phi1Std, pt1);
+        registry.fill(HIST("hPhiByVtxPhi"), phi1ByVtx, pt1); //trick to have correct Phi range
+        registry.fill(HIST("hPhiDifferenceTwoMethods"), getDeltaPhi(phi1ByVtx, phi1Std), pt1);
+
+        //get corresponding gen-level D0, if exists, and evaluate gen-rec phi-difference with two approaches
+        if (std::abs(candidate1.flagMCMatchRec()) == 1 << D0ToPiK) {                                                     //ok to keep both D0 and D0bar
+          int indexGen = RecoDecay::getMother(particlesMC, candidate1.index0_as<aod::BigTracksMC>().label(), 421, true); //MC-gen corresponding index for MC-reco candidate
+          if (indexGen > 0) {
```

Comment:
What I want to do is to enaure that what follows is done only if a gen-reco match is ensured, but looking how the returned variable is initialised in RecoDecay::getMother, I think I should set the condition to > -1, rather than 0 (this is what I'm doing now, please correct me if not ok)

Reply:
The gen-rec match is ensured already by requiring:
```c++
if (std::abs(candidate1.flagMCMatchRec()) == 1 << D0ToPiK) {   
```
Therefore `getMother` will always return a valid index.

Reply:
Ok, removed the condition